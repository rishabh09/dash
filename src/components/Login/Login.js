import React from 'react'
import {connect} from 'react-redux'
import { googleAuth } from '../../actions/google-auth';

export class Login extends React.Component {
  googleSignin = () => {
    this.props.dispatch(googleAuth())
  }

  render () {
    return (
      <div> {
        this.props.auth.login ? 
        this.props.auth.code :
        <button onClick={this.googleSignin}>Login With Google</button>
      }
      </div>
    )
  }
}


export default connect(({auth}) => {return {auth}})(Login)