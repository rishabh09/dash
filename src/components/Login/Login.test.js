import React from 'react'
import {shallow} from 'enzyme'
import {Login} from './Login'

describe('Component: Login', () => {
  it('should render component properly', () => {
    const props = {
      auth : {}
    }
    const wrapper = shallow(<Login {...props}/>)
    expect(wrapper).toMatchSnapshot()
  })
})