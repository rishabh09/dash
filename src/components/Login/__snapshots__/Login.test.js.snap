// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Login should render component properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    auth={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        <button
          onClick={[Function]}
        >
          Login With Google
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login With Google",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Login With Google",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          <button
            onClick={[Function]}
          >
            Login With Google
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login With Google",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Login With Google",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
